File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row = set()
        col = set()
        
        for r in range(0,len(matrix)):
            for c in range(0, len(matrix[r])):
                if matrix[r][c] == 0:
                    row.add(r)
                    col.add(c)
        
        for r in range(0,len(matrix)):
            for c in range(0, len(matrix[r])):
                if r in row or c in col:
                    matrix[r][c] = 0
        

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function(matrix) {
    const row = new Set();
    const col = new Set();
    
    for (let r = 0; r < matrix.length; r++) {
        for (let c = 0; c < matrix[r].length; c++){
            if (matrix[r][c] === 0) {
                row.add(r);
                col.add(c)
            }
        }
    }
  
    for (let r = 0; r < matrix.length; r++) {
        for (let c = 0; c < matrix[r].length; c++) {
            if (row.has(r) || col.has(c)) matrix[r][c] = 0
        }
    }
};